cdap-master-spi/src/main/java/io/cdap/cdap/master/spi/namespace/NamespaceLister.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.master.spi.namespace;
+
+import java.util.List;
+
+/**
+ * Interface for listing details of CDAP namespaces.
+ */
+public interface NamespaceLister {
```

Comment:
this sounds like the implementation is listing namespaces. It also makes it very specific to namespace lists, which would be awkward if we ever want to extend it. I think something like NamespaceLifecycle or NamespaceListener something would be more clear.

I also think it would be better to deprecate the onNamespaceCreation and onNamespaceDeletion methods in MasterEnvironment and create corresponding onNamespaceCreation(NamespaceDetail) and onNamespaceDeletion(NamespaceDetail) methods with default no-op implementations into this interface.

Reply:
Also, the onNamespaceCreation and onNamespaceDeletion methods should have javadocs that describe the contract in more detail. For example, they need to be idempotent operations. If onNamespaceCreation() throws an exception, the CDAP namespace will not be created. If onNamespaceDeletion() throws an exception, what happens to the CDAP namespace delete operation?

Reply:
fixed